lib/monocle.js

```diff
@@ -5,13 +5,17 @@ var ResourceCache = require('./cache/resource');
 var querystring = require('querystring');
 var CollectionCache = require('./cache/collection');
 
-function Monocle(http) {
+function Monocle(http, promise) {
     this._http = http;
     this._base = '/';
     this._cache = new Store(new ResourceCache('monocle', { capacity: 100 }));
     this._batched = [];
     this._batchTimeout = null;
     this._queuedGets = {};
+    // If a non-native Promise library is being used, assume it is already available to us
+    if (promise) {
+        Promise = promise;
```

Comment:
probably should assign it to `this`, because as written it will replace a global variable.